src/services/swork.ts

```diff
@@ -1,35 +1,82 @@
 import {ApiPromise} from '@polkadot/api';
-import {Request, Response} from 'express';
+import {Request} from 'express';
 import {KeyringPair} from '@polkadot/keyring/types';
+import {sendTx, queryToObj} from './util';
 
+/**
+ * Send extrinsics
+ */
 export async function register(
   api: ApiPromise,
   krp: KeyringPair,
-  req: Request,
-  res: Response
+  req: Request
 ) {
-  return await api.tx.swork
-    .register(
-      req.body['ias_sig'],
-      req.body['ias_cert'],
-      req.body['account_id'],
-      req.body['isv_body'],
-      '0x' + req.body['sig']
-    )
-    .signAndSend(krp, ({events = [], status}) => {
-      console.log(`Current status is ${status.type}`);
+  const tx = api.tx.swork.register(
+    req.body['ias_sig'],
+    req.body['ias_cert'],
+    req.body['account_id'],
+    req.body['isv_body'],
+    '0x' + req.body['sig']
+  );
 
-      // TODO: Extract to util function
-      if (status.isFinalized) {
-        console.log(`Transaction included at blockHash ${status.asFinalized}`);
-        // Loop through Vec<EventRecord> to display all events
-        events.forEach(({phase, event: {data, method, section}}) => {
-          console.log(`\t' ${phase}: ${section}.${method}:: ${data}`);
-        });
+  return await sendTx(tx, krp);
+}
+
+export async function reportWorks(
+  api: ApiPromise,
+  krp: KeyringPair,
+  req: Request
+) {
+  const tx = api.tx.swork.reportWorks(
+    '0x' + req.body['pub_key'],
+    '0x' + req.body['pre_pub_key'],
+    req.body['block_height'],
+    '0x' + req.body['block_hash'],
+    req.body['reserved'],
+    req.body['files_size'],
+    req.body['added_files'].map((file: any) => {
+      const rst: [string, number] = ['0x' + file.hash, file.size];
+      return rst;
+    }),
+    req.body['deleted_files'].map((file: any) => {
+      const rst: [string, number] = ['0x' + file.hash, file.size];
+      return rst;
+    }),
+    '0x' + req.body['reserved_root'],
+    '0x' + req.body['files_root'],
+    '0x' + req.body['sig']
+  );
+
+  return await sendTx(tx, krp);
+}
+
+/**
+ * Queries
+ */
+export async function identity(api: ApiPromise, req: Request) {
+  const address = req.query['address'];
+  const pks: string[] = queryToObj(await api.query.swork.idBonds(address));
+  return await Promise.all(
```

Comment:
consider adding limiting max number of parallels calls here?

Reply:
Add TODO cause currently we only has 10 wr